src/libutil/compression.cc

```diff
@@ -214,14 +281,21 @@ void decompress(const std::string & method, Source & source, Sink & sink)
         return decompressBzip2(source, sink);
     else if (method == "br")
         return decompressBrotli(source, sink);
+#if HAVE_ZSTD
```

Comment:
Why does this need CPP guards, but brotli doesn't?

Reply:
Because nix's brotli support is always "possible"-- if not built with libbrotli Nix will look for the brotli executable on PATH and try to use that with fingers crossed.  Among other disadvantages, the way this is implemented operates on the entire input at once IIRC, reading into a giant std::string.  Doing this for zstd didn't seem like a good idea, thinking it should be supported properly or not at all.

Brotli actually has some preprocessor bits but they're inside the decompressBrotli method instead of around it.